spec/routes/web/admin/admin_spec.rb

```diff
@@ -682,6 +682,49 @@ def page_data
     ENV.delete("DONT_RAISE_ADMIN_ERRORS")
   end
 
+  it "supports setting quota of Project" do
+    p = Project.create(name: "Default")
+
+    fill_in "UBID or UUID", with: p.ubid
+    click_button "Show Object"
+    expect(page.title).to eq "Ubicloud Admin - Project #{p.ubid}"
+
+    click_link "Set Quota"
+    path = page.current_path
+
+    # Create a new quota
+    select "VmVCpu", from: "resource_type"
+    fill_in "value", with: 512
+    click_button "Set Quota"
+    expect(page).to have_flash_notice("Set quota")
+    expect(page.title).to eq "Ubicloud Admin - Project #{p.ubid}"
+    expect(p.effective_quota_value("VmVCpu")).to eq(512)
+
+    # Update existing quota
+    visit path
+    select "VmVCpu", from: "resource_type"
+    fill_in "value", with: 1024
+    click_button "Set Quota"
+    expect(page).to have_flash_notice("Set quota")
+    expect(p.effective_quota_value("VmVCpu")).to eq(1024)
+
+    # Set quota to zero
+    visit path
+    select "VmVCpu", from: "resource_type"
+    fill_in "value", with: 0
+    click_button "Set Quota"
+    expect(page).to have_flash_notice("Set quota")
+    expect(p.effective_quota_value("VmVCpu")).to eq(0)
+
+    # Remove quota when value is blank
+    visit path
+    select "VmVCpu", from: "resource_type"
+    fill_in "value", with: ""
+    click_button "Set Quota"
+    expect(page).to have_flash_notice("Set quota")
+    expect(p.effective_quota_value("VmVCpu")).to eq(32)
```

Comment:
Update the specs to test for `nil` value handling when there is no existing quota:

```suggestion
    2.times do
      # Remove quota when value is blank
      # Ensure value being blank doesn't add quota
      visit path
      select "VmVCpu", from: "resource_type"
      fill_in "value", with: ""
      click_button "Set Quota"
      expect(page).to have_flash_notice("Set quota")
      expect(p.effective_quota_value("VmVCpu")).to eq(32)
    end
```